docs/home/advanced_usage.md

```diff
@@ -0,0 +1,98 @@
+
+
+## Programming A Local Drive
+
+In the [Getting Started](getting_started.md) section, you might have noticed that we used `...uniform...` when defining the various drives applied to the atoms. This syntax refers to one of three ways of expressing the spatial modulation of the waveform. What do I mean by spatial modulation? In this case, you think of a drive having a temporal component, e.g., the waveform, and a spatial part, which we call spatial modulation. The spatial modulation is a scale factor applied to the waveform when acting on a particular site. For example, if I have two atoms and I have a spatial modulation for site-`0` is `0.1` and for site-`1` is `0.2`, then the waveform will be scaled by `0.1` when acting on-site `0` and `0.2` when applied to site-`1`.
+
+Intuitively, `uniform` spatial modulation simply means that regardless of the atom, the waveform value is always the same. The other two options for defining spatial modulations are: `scale` and `location`. They have two distinct use cases:
+
+## `scale` Method
+
+When calling the 'scale ' method, you can pass either a list of real values or a string. The list of values defines the scaling for every atom in the system, so the number of elements must equal the number of sites in your geometry. The string is a placeholder that allows you to define that mask as a list of values by passing them in through `assign`, `batch_assign`, or `args`. For `assign` and `batch_assign`, you can pass a list or a list of lists, respectively. On the other hand, `args`, the list of values, must be inserted into the `args` tuple when calling `run` or `run_async`, for example, let's take a two-atom program:
+
+
+```python
+from bloqade import start
+
+program = (
+    start.add_position([(0, 0), (0, 5)])
+    .rydberg.detuning.scale("local_detuning")
+    .piecewise_linear([0.1, 1.0, 0.1], [0, "detuning", "detuning", 0])
+    .amplitude.uniform.piecewise_linear([0.1, 1.0, 0.1], [0, 15, 15, 0])
+    .args(["local_detuning", "detuning"])
+    .bloqade.python()
+)
+```
+
+To call the `run` method properly, you must unpack the spatial modulation list into the tuple, e.g.
+
+```python
+local_detuning = [0.4, 0.1]
+
+program.run(100, args=(*local_detuning, 30))
+```
+
+## `location` Method
+
+The `location` method takes two arguments; the second is optional. The first argument is an integer or a list of integers corresponding to the site(s) to which you want to apply the spatial modulation. The second argument is the value you wish to apply to the site(s). If you pass a list of integers, you must pass a list of values for the second argument, while if you pass a single integer, you must pass a single value. The interpretation is that for any sites not specified in the first argument, the spatial modulation is `0.0`. Note that the values you pass in can be a real value, a string (for a variable), or a scalar expression. Below is an example using `location`:
```

Comment:
Wouldn't the spatial modulation be 1.0? Otherwise this seems to imply we do some aggressive pre-emptive masking